hs/app/reach/Main.hs

```diff
@@ -2450,6 +2451,199 @@ whoami = command "whoami" $ info f fullDesc
   where
     f = pure . script $ write [N.text| echo "$whoami'" |]
 
+-- https://stackoverflow.com/a/60790430
+newtype GitHubGistResponse = GitHubGistResponse String
+instance FromJSON GitHubGistResponse where
+  parseJSON = withObject "GitHubGistResponse" $ \o -> GitHubGistResponse <$> o .: "html_url"
+
+support :: Subcommand
+support = command "support" $ info (pure step1) d
+  where
+    d = progDesc "Upload index.rsh and index.mjs to help us troubleshoot!"
+    splitByAmpersands :: BSLC8.ByteString -> [Text]
+    splitByAmpersands s = T.splitOn "&" (pack $ BSLC8.unpack s)
+    splitByEqualsSigns :: Text -> [Text]
+    splitByEqualsSigns s = T.splitOn (pack "=") s
+    process :: BSLC8.ByteString -> [[Text]]
+    process gitHubResponseString = map splitByEqualsSigns
+      $ splitByAmpersands gitHubResponseString
+    isDeviceCodePair :: [Text] -> Bool
+    isDeviceCodePair pair = head pair == pack "device_code"
+    isErrorPair :: [Text] -> Bool
+    isErrorPair pair = head pair == pack "error"
+    isUserCodePair :: [Text] -> Bool
+    isUserCodePair pair = head pair == pack "user_code"
+    isAccessTokenPair :: [Text] -> Bool
+    isAccessTokenPair pair = head pair == pack "access_token"
+    clientId :: String
+    clientId = "b0e24d4cc8251c6cd14c"
+    scope :: String
+    scope = "gist"
+    language :: String
+    language = "JavaScript"
+    typeForUploadJson :: String
+    typeForUploadJson = "application/javascript"
+    acceptHeader = "Accept"
+    authorizationHeader = "Authorization"
+    step1 :: App
+    step1 = do
+      let dataJson =
+            object
+              [ "client_id" .= clientId
+              , "scope" .= scope
+              ]
+      parsedRequest <- parseRequest "POST https://github.com/login/device/code"
+      let request = setRequestBodyJSON dataJson parsedRequest
+      response <- httpLBS request
+      let githubResponseString = getResponseBody response
+      let arrayOfArrayOfText = process githubResponseString
+      let deviceCodePair = head $ filter isDeviceCodePair arrayOfArrayOfText
+      let deviceCode = deviceCodePair !! 1
+      let userCodePair = head $ filter isUserCodePair arrayOfArrayOfText
+      let userCode = userCodePair !! 1
+      liftIO . T.putStrLn $ pack $ "Your user code is " ++ unpack userCode
+      liftIO . T.putStrLn $ pack "Please enter it at https://github.com/login/device"
+      liftIO . T.putStrLn $ pack ""
+      liftIO . T.putStrLn $ pack "Type 'y' after successful authorization"
+      liftIO . T.putStrLn $ pack "to upload index.mjs, index.rsh, or both:"
+      userEnteredCharacter <- liftIO getChar
+      case toUpper userEnteredCharacter of
+        'Y' -> do step2WithThe deviceCode
+        _ -> do
+          liftIO . T.putStrLn $ pack ""
+          liftIO . T.putStrLn $
+            "No files uploaded; run reach support again to make another try."
+
```

Comment:
You still have lots of unnecessary newlines

Reply:
[f2065ce](https://github.com/reach-sh/reach-lang/pull/762/commits/f2065ce634207ef5829f112fb6963e42051120cf) should address this.